oclc/classify.py

```diff
@@ -769,6 +955,72 @@ def lookup_by(self, **kwargs):
         representation, cached = Representation.get(self._db, url)
         return representation.content
 
+class OCLCLookupCoverageProvider(MetadataWranglerBibliographicCoverageProvider):
+
+    def __init__(self, collection, api=None, **kwargs):
+        super(OCLCLookupCoverageProvider, self).__init__(
+            collection, registered_only=False, **kwargs
+        )
+        self.api = api or OCLCClassifyAPI(self._db)
+
+
+class IdentifierLookupCoverageProvider(OCLCLookupCoverageProvider):
+    """Does identifier (specifically, ISBN) lookups using OCLC Classify.
+    """
+    SERVICE_NAME = "OCLC Classify Identifier Lookup"
+    INPUT_IDENTIFIER_TYPES = [Identifier.ISBN]
+    DATA_SOURCE_NAME = DataSource.OCLC
+
+    parser = OCLCClassifyXMLParser()
+
+    def _get_tree(self, **kwargs):
+        xml = self.api.lookup_by(**kwargs)
+        return etree.fromstring(xml, parser=etree.XMLParser(recover=True))
+
+    def process_item(self, identifier):
+        """Ask OCLC Classify about a single ISBN and create an Edition
+        based on what it says.
+        """
+        # Perform a title/author lookup.
+        metadata_list = []
+        failure = None
+
+        try:
+            tree = self._get_tree(isbn=identifier.identifier)
+            code, owi_data = self.parser.initial_look_up(self._db, tree)
+            if code in [self.parser.SINGLE_WORK_DETAIL_STATUS, self.parser.SINGLE_WORK_SUMMARY_STATUS]:
+                metadata_list = self._single(self._db, tree, identifier)
+            elif code == self.parser.MULTI_WORK_STATUS:
+                metadata_list = self._multiple(self._db, owi_data, identifier)
+            elif code == self.parser.NOT_FOUND_STATUS:
+                return CoverageFailure(identifier, traceback.format_exc(), data_source=DataSource.OCLC, transient=True)
+            if metadata_list:
+                for metadata in metadata_list:
+                    self._apply(metadata, identifier)
+            return identifier
+
+        except IOError as e:
+            return self.failure(identifier, e.message)
+
+    def _single(self, _db, tree, identifier):
+        return [self.parser.parse(self._db, tree, [identifier])]
+
+    def _multiple(self, _db, owi_data, identifier):
+        results = []
+        for item in owi_data:
+            tree_from_owi = self._get_tree(owi=item.identifier)
+            results.append(self.parser.parse(self._db, tree_from_owi, [identifier, item]))
+
+        return results
+
+    def _apply(self, metadata, identifier):
+        edition = self.edition(identifier)
+        work = self.work(edition.primary_identifier)
+        metadata.apply(
+            edition, collection=None #, replace=self.replacement_policy
```

Comment:
You can take out the commented part of this line; it turns out we don't need it, I'm pretty sure.